drakrun/drakrun/main.py

```diff
@@ -565,18 +571,45 @@ def log_startup_failure(self, log_path):
                     self.log.info("Injection failed with error: %s", entry["Error"])
                     break
 
+    def _memory_dump(self, vm_name, dump_filepath):
+        """Dump raw memory from running vm using vmi-dump-memory and compress it with gzip"""
+
+        with tempfile.NamedTemporaryFile() as raw_memdump, open(
+            f"{dump_filepath}.gz", "wb"
+        ) as compressed_file:
+            self.log.info(f"Dumping raw memory from {vm_name} guest...")
+            try:
+                dump_args = ["vmi-dump-memory", vm_name, raw_memdump.name]
+                subprocess.run(dump_args, check=True)
+            except subprocess.CalledProcessError as e:
+                self.log.error(f"Dumping raw memory from {vm_name} failed.")
+                raise e
+
+            self.log.info(f"Compressing {vm_name} guest memory dump...")
+            subprocess.check_call(
+                ["gzip", "-c", raw_memdump.name], stdout=compressed_file
+            )
+
+            self.log.info(f"Raw memory dump from {vm_name} succeeded.")
+
     def analyze_sample(self, sample_path, workdir, outdir, start_command, timeout):
         analysis_info = dict()
 
         dns_server = self.config.config["drakrun"].get("dns_server", "8.8.8.8")
         drakmon_log_fp = os.path.join(outdir, "drakmon.log")
+        raw_memory_dump = self.config.config["drakrun"].getboolean(
+            "raw_memory_dump", fallback=False
+        )
 
         with self.run_vm() as vm, graceful_exit(
             start_dnsmasq(self.instance_id, dns_server)
         ), graceful_exit(start_tcpdump_collector(self.instance_id, outdir)), open(
             drakmon_log_fp, "wb"
         ) as drakmon_log:
 
+            if raw_memory_dump:
+                self._memory_dump(vm.vm_name, outdir, "memory_dump_pre_sample.raw")
```

Comment:
As I've mentioned, doing pre-analysis dumps seems redundant. We can solve this by exporting a snapshot.

Reply:
Yes, sorry for the inconvenience, this lane is outdated and was supposed to be removed with the exporting commit.

Probably tomorrow I will push the exporting demo functionality. I think the convenient way will be for you to validate what are the adjustments that are necessary, and if there will be simply too many of them, you can freely implement exporting by yourself : )